src/datasources/balances-api/zerion-balances-api.service.ts

```diff
@@ -354,4 +354,9 @@ export class ZerionBalancesApi implements IBalancesApi {
     );
     if (current > this.limitCalls) throw new LimitReachedError();
   }
+
+  private _checkFiatCode(fiatCode: string): void {
+    if (!this.fiatCodes.includes(fiatCode.toUpperCase()))
+      throw new DataSourceError(`Unsupported currency code: ${fiatCode}`, 400);
+  }
```

Comment:
Nit: I wouldn't abstract this as the call is somewhat hard to follow. What do you think?